FIx - removed html from comparison strings

AnimusMachina opened the pull request:
The natural sorting plugin doesn't remove html before comparing the strings with each other.

I think this is unintended behaviour as adding some tags (e.g. adding links to table entries) can mess up the ordering.

DataTables commented:
The problem with doing it that way is that is can be really slow. It might also result in jQuery errors if your cell content isn't all wrapped in HTML tags (i.e. you might pass plain text into `$()` resulting in an exception).

The only way I'm really found to do this thus far is to strip the HTML with regex, which is far from perfect, but it gets it right in nearly all cased.

Happy to pull the PR in if you would be willing to update it to do that?

AnimusMachina commented:
Sorry for repsonding so late :D

I changed the html stripping to regex, but it might break especially if there are broken html tags involved. I tried to add a configuration variable that lets users choose to strip html or not but i don't know if there is a way to pass configuration variables directly to the sorting algorithm so instead i added 'natural-nohtml' to the sort extension.

If there is an easier/better way to do it or if this is not desired at all i can change/remove it.

DataTables commented:
Looks good to me - thanks for the update! Are you happy for this to be included under the MIT license?

Also do you want to add an `@author` tag for your own work on this? :-)

AnimusMachina commented:
I don't have any problems with the license and would be happy to include my work under it :)

As for the @author tag should i write a short description of what i changed or something like that?

I'm not terribly experienced with all that.

DataTables commented:
Just `@author {yourname|username|etc}` will do. Git contains the histroy so there is no need to store that in plain text.

AnimusMachina commented:
Aaaand done (i think :D)

Thank you for your help :)

DataTables commented:
No - thank you for your contribution! :-)
